### PR TITLE
send notifications through fcm on both platforms

### PR DESCRIPTION
**Pull request by upacyxou**
- add sending messages through fcm on both platforms

### CODE REVIEW

File: src/helpers/sendFirebaseNotification.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const messaging = getMessaging(firebase)
+
+export default function (token: string, title?: string, postId?: number) {
+  const message = {
+    token,
+    android: {
+      priority: 'high',
+    },
+    data: {
+      postId: String(postId),
```
Comment (smyzin) [inline src/helpers/sendFirebaseNotification.ts:13]: If no `title` exists in `runNotificationListener`, `postId` will be equal to `undefined`. And on this line, you are converting `undefined` to string. Is it correct?
Should we create a data const equal to an empty object if `postId === undefined` and `{ postId: String(postId) }` if postId exists?

File: src/helpers/runNotificationListener.ts [hunk 2]
```diff
@@ -18,24 +19,31 @@ const rootFeeds: { [key: number]: string } = env.isProduction
...
-obssContract.on('FeedPostAdded', async (feedId) => {
-  const title = rootFeeds[feedId.toNumber()]
-    ? `Someone posted at ${rootFeeds[feedId.toNumber()]}`
-    : undefined
+obssContract.on(
+  'FeedPostAdded',
+  async (feedId: BigNumber, postID: BigNumber) => {
+    const title = rootFeeds[feedId.toNumber()]
+      ? `Someone posted at ${rootFeeds[feedId.toNumber()]}`
+      : undefined
```
Comment (smyzin) [inline src/helpers/runNotificationListener.ts:27]:
```
const feed = rootFeeds[feedId.toNumber()]
const title = feed && `Someone posted at ${rootFeeds[feedId.toNumber()]}`
```
?

**Review by smyzin — APPROVED** [review]
Looks clear

File: src/helpers/runNotificationListener.ts (hunk 2 above)
Comment (T-Damer) [reply src/helpers/runNotificationListener.ts:27]:
```TS
const feed = rootFeeds[feedId.toNumber()] const title = feed && `Someone posted at ${feed}`
```

File: src/helpers/sendFirebaseNotification.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      payload: {
+        aps: {
+          contentAvailable: true,
+        },
+      },
+      headers: {
+        'apns-push-type': 'background',
+        'apns-priority': '5',
+        'apns-topic': 'xyz.ketl',
+      },
```
[the remark this answers is not in the document]
Comment (upacyxou) [reply src/helpers/sendFirebaseNotification.ts:32]: nah, we won't re-use it anywhere and we will not change it in the future

File: src/helpers/runNotificationListener.ts [hunk 4]
```diff
@@ -18,24 +19,31 @@ const rootFeeds: { [key: number]: string } = env.isProduction
...
+        } else {
+          // FCM token
+          await sendFirebaseNotification(
+            token,
+            title,
+            title ? postID.toNumber() : undefined
+          )
+        }
+      } catch (err) {
+        console.error(err)
```
[the remark this answers is not in the document]
Comment (upacyxou) [reply src/helpers/runNotificationListener.ts:44]: nah, I don't think so, there'll a lot of errors about missing entity when users will start deleting/reinstalling app

**Review by T-Damer — APPROVED** [review]

**Pull request merged**